### DISCUSSION TITLE
[Idea] Invoice filename to use Order Reference rather than Invoice Number

### DISCUSSION THREAD

**Discussion by extrasgroup** [PrestaShop/PrestaShop#37251] [Ideas]
### The problem

When an orders is created in Prestashop a reference ID comprising of 9 Characters e.g. ABCDEFGHI is created to identify it.

Prestashop also creates and Order ID (and an Invoice ID)

I presume this 9 char ref was created so as to not give away to the customer how many orders in total have been placed on the shop. I think is a good idea, opposed to using the Order/Invoice number which increments by one every order. 

I personally have updated our PDF templates to only use this 9 char reference.

Currently when Prestashop generates invoice paperwork (PDF) to send the the customer the invoice ID is used as the PDFs filename. Which is not ideal if you are trying to hide this from the customer.

### The solution

Offer an option in the Shop Parameters > Order Settings to use either the invoice ID or the order reference when generating paperwork.

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by Hlavtox** [comment]
People don't care about your invoice number, and you won't care too when your shop grows.

It's a standard to number invoices in a standardized order. Doing it randomly or missing some numbers is just going to draw attention of the financial bureau on you, because they could sense you could be deleting some invoices.
